aten/src/ATen/FunctionalTensorWrapper.h

```diff
@@ -172,6 +182,8 @@ struct TORCH_API FunctionalTensorWrapper : public c10::TensorImpl {
   // change the value tensor that it points to over time.
   Tensor value_;
   int64_t level_;
+  uint64_t mutation_counter_ = 0;
+  uint64_t mutation_from_triton_kernels_counter_ = 0;
```

Comment:
actually, maybe we should change this name (and its corresponding helpers) to `mutation_hidden_from_autograd_counter_`?

(1) It conveys the true intent - the idea is that if we have an input mutation that is hidden from autograd, then once we convert the input mutation to a `copy_()` we know it will be safe to hide the `copy_()` from autograd as well.

(2) it's more general - maybe in the future there will be some other mutations that we know are safe to hide from autograd, so we can reuse this infra.

And do you mind adding a comment explaining this?

Reply:
+1